src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Logging/DocumentationSignatureGenerator.PartVisitor.cs

```diff
@@ -218,6 +220,18 @@ private static string GetEscapedMetadataName(IMemberDefinition member)
                 return name.Replace('<', '{').Replace('>', '}');
             }
 #endif
+
+            private static TypeDesc InstantiateContainingType(DefType type)
+            {
+                DefType containingType = type.ContainingType;
+                if (containingType is null)
+                    return null;
+
+                if (!containingType.HasInstantiation)
+                    return containingType;
+
+                return containingType.InstantiateAsOpen().InstantiateSignature(type.Instantiation, default(Instantiation));
```

Comment:
This will crash if the type doesn't follow the C# scheme where nested types have at least the same generic arity as their containing type.

Reply:
Honestly this was me trying to figure out how to do things, and this seemed to work. So the question is:
Given a containing type, which will have signature variables, how do I create an instantiation without signature variables given that I have the nested type which is fully instantiated.

And a sub question - is there a helper to determine if type is fully instantiated (no signature variables)? (for assert).

Reply:
You can either call `MakeInstantiatedType` on the generic definition (that one is faster, more obvious and usually preferred), or just use the `InstantiateSignature` like you do here. The latter is fewer lines of code here because you don't need to create a precisely-sized instantiation (supplying larger one is fine).

There is a way to check if there's any signature variables left, but here you're guaranteed there won't be. `InstantiateSignature` replaces them all, so you'd only have them if `type.Instantiation` had any in the first place.

```suggestion
                // Does the type follow C# scheme where nested types inherit generic parameters from their containing type?
                if (containingType.Instantiation.Length > type.Instantiation.Length)
                    return containingType;
                
                return containingType.InstantiateAsOpen().InstantiateSignature(type.Instantiation, default(Instantiation));
```

Reply:
It can be written in il. The nested type "inheriting" generic parameters of the owning type is a C# thing. In il it's fine if the owning type has one generic parameter and nested has zero. We'll probably make a garbage documentation name for it but at least not crash. This is why I'm pushing reflection notation everywhere.